### PR TITLE
Adding Channel to documentation

### PR DESCRIPTION
**Pull request by hkateu**
Hi, 
I added documentation for Channel based on [issue2866](https://github.com/typelevel/fs2/issues/2866). 

I also think that for new users to easily follow the documentation, the concurrency primitives, we can do the following: 

- Change the order  in the docs to: `Signal` , `Channel`, `Topic` then `Queue`. Going from a `Channel` to a `Topic` seems logical since its a  many-to-one vs many to many pub-sub relationship.
- Show that Topics are used for a many-to-many pub-sub scenario, since it's not explicitly mentioned in the docs.

Any feedback is welcome.

### CODE REVIEW

File: testdata/tom_sawyer.txt
Comment (mzuehlke) [inline testdata/tom_sawyer.txt:1]: why did you add this file ?

Comment (hkateu) [reply testdata/tom_sawyer.txt:1]: I have no idea how that got generated, but I've taken it out. Submitting changes.

File: site/concurrency-primitives.md [hunk 1]
```diff
@@ -3,12 +3,31 @@
...
+`Channel` implements a publish-subscribe pattern and is particularly useful where we have multiple `publishers` and a single `subscriber`. In the following example, we have two streams, `pub1` and `pub2` publishing the strings `"Hello"` and `"World"` every `1` and `2` seconds respectively. Additionally, we have `sub`, a subscriber that consumes and prints each element. The three streams are then run in parallel and interrupted after `6` seconds.
+
+```scala mdoc:silent
+import cats.effect._
+import fs2.Stream
+import scala.concurrent.duration._
+import cats.effect.unsafe.implicits.global
+import fs2.concurrent.Channel
+
+Channel.unbounded[IO, String].flatMap{channel =>
```
Comment (mpilquist) [inline site/concurrency-primitives.md:23]:
```suggestion
Channel.unbounded[IO, String].flatMap { channel =>
```
